packages/data-provider/src/services/submissionService.ts

```diff
@@ -101,33 +112,42 @@ const service = (dependencies: BaseDependencies) => {
 
 		submittedDataToValidate?.forEach((data) => entitiesToProcess.add(data.entityName));
 
-		const insertsToValidate =
-			submission.data?.inserts &&
-			Object.entries(submission.data.inserts).reduce<NewSubmittedData[]>((acc, [entityName, submissionData]) => {
-				entitiesToProcess.add(entityName);
-				acc.push(
-					...submissionData.records.map((record) => {
-						const newSubmittedData: NewSubmittedData = {
-							data: record,
-							dictionaryCategoryId: categoryId,
-							entityName,
-							isValid: false, // By default New Submitted Data is created as invalid until validation process proves otherwise
-							organization: submission.organization,
-							originalSchemaId: submission.dictionaryId,
-							lastValidSchemaId: submission.dictionaryId,
-							systemId: generateIdentifier(entityName, record),
-							createdBy: userName,
-						};
-						return newSubmittedData;
-					}),
-				);
-				return acc;
-			}, []);
+		const insertsToValidate = submission.data?.inserts
+			? Object.entries(submission.data.inserts).flatMap(([entityName, submissionData]) => {
+					entitiesToProcess.add(entityName);
+
+					return submissionData.records.map((record) => ({
+						data: record,
+						dictionaryCategoryId: categoryId,
+						entityName,
+						isValid: false, // By default, New Submitted Data is created as invalid until validation proves otherwise
+						organization: submission.organization,
+						originalSchemaId: submission.dictionaryId,
+						lastValidSchemaId: submission.dictionaryId,
+						systemId: generateIdentifier(entityName, record),
+						createdBy: userName,
+					}));
```

Comment:
The `commit` operation verifies that the Submission is valid.  However, instead of setting `lastValidSchemaId` before validating I've adjusted the code to set it after validation. That way when `lastValidSchemaId` is null and `isValid` is false means data has never been validated or never valid.  Otherwise `lastValidSchemaId` will be set when validation has passed.